builtin/aws/alb/releaser.go

```diff
@@ -467,6 +856,100 @@ func (r *Releaser) Status(
 	return &report, nil
 }
 
+// Destroy will modify or delete Listeners, so that the platform can destroy the
+// target groups
+func (r *Releaser) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: release.Region,
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+
+	rm := r.resourceManager(log)
+
+	// If we don't have resource state, this state is from an older version
+	// and we need to manually recreate it. Older versions did not save anything
+	// for Security Group Ids
+	if release.ResourceState == nil {
+		rm.Resource("security_group").SetState(&Resource_SecurityGroup{})
+		rm.Resource("load_balancer").SetState(&Resource_LoadBalancer{
+			Arn: release.LoadBalancerArn,
+			// resourceRecordSetDestroy relies on the Load Balancer state having the
+			// Zone Id present.
+			ZoneId: r.config.ZoneId,
+		})
+		l, err := findListenerArnFromTargetGroup(sess, log, release.TargetGroupArn, release.LoadBalancerArn)
+		if err != nil {
+			return err
+		}
+		rm.Resource("listener").SetState(&Resource_Listener{
+			TgArn: release.TargetGroupArn,
+			Arn:   l,
+		})
+		// older releases don't have any security group information
+		rm.Resource("record_set").SetState(&Resource_RecordSet{
+			FQDN: r.config.FQDN,
+		})
+	} else {
+		// Load our set state
+		if err := rm.LoadState(release.ResourceState); err != nil {
+			return err
+		}
+	}
+
+	// Destroy All
+	return rm.DestroyAll(ctx, log, sg, ui, sess)
+}
+
+func findListenerArnFromTargetGroup(
+	sess *session.Session,
+	log hclog.Logger,
+	tgArn string,
+	lbArn string,
+) (string, error) {
+
+	var listener *elbv2.Listener
+
+	elbsrv := elbv2.New(sess)
+	out, err := elbsrv.DescribeListeners(&elbv2.DescribeListenersInput{
+		LoadBalancerArn: &lbArn,
+	})
+	if err != nil {
+		return "", err
+	}
+
+LLOOP:
+	// We don't know the ARN of the listener, only the Target Group and the
+	// LoadBalancer. We need to loop through each Listener's actions to find the
+	// target group that matches.
+	for j, ol := range out.Listeners {
+		for _, def := range ol.DefaultActions {
+			if def.ForwardConfig != nil {
+				for _, tg := range def.ForwardConfig.TargetGroups {
+					if *tg.TargetGroupArn == tgArn {
+						listener = out.Listeners[j]
+						break LLOOP
+					}
+				}
+			}
+		}
+	}
+
+	if listener != nil {
+		return *listener.ListenerArn, nil
+	}
+	return "", fmt.Errorf("unable to find listener")
```

Comment:
Since we have grpc's `codes` and `status` funcs, it's nice to return a more detailed error here. So instead of a `fmt.Errorf` we could return a new `status.Error` with `codes.NotFound` as the error code